### DISCUSSION TITLE
ClickHouse: Nullable(DateTime) as time field throwing error in chart UI

### DISCUSSION THREAD

**Discussion by ravikishorer** [apache/superset#17253] [Q&A / Help]
ClickHouse: Nullable(DateTime) as time field throwing error in create a new chart UI as the query it is generating contains time fields which have milli second precision and clickhouse couldn't convert them to date time.

### Expected results

It should work correctly.

### Actual results

Create chart area shows error when time field is present.

#### Screenshots

![image](https://user-images.githubusercontent.com/83571469/118534382-cb486a80-b766-11eb-91ec-c0d9bf1d368f.png)

The field is identified correctly as a temporal field in the fields section of the chart UI
![image](https://user-images.githubusercontent.com/83571469/118535362-00a18800-b768-11eb-9b00-8fee953f2a70.png)

#### How to reproduce the bug

1. Clickhouse create table
```
CREATE TABLE trainingReport
 ( 
Training_ID Int32,
date Nullable(DateTime),
) ENGINE = MergeTree()
 ORDER BY tuple();
```
2. Try to create a chart with this dataset which contains a temporal field value

### Environment
- superset version: 1.1.0
- python version: 3.8.5
- node.js version: 12.18.3

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context
Query is becoming something like 
`SELECT COUNT(*) AS count
FROM "default"."trainingReport"
WHERE "Last_Published_Date__UTC_TimeZone_" >= '2021-05-10 00:00:00.000000'
  AND "Last_Published_Date__UTC_TimeZone_" < '2021-05-17 00:00:00.000000'
ORDER BY count DESC
LIMIT 10000;`

https://github.com/apache/superset/blob/master/superset/db_engine_specs/clickhouse.py#L66 is the issue as the field type is Nullable(DateTime), no proper conversion is happening for the **from** and **to** dates as the function is simply checking for utils.TemporalType.DATE and utils.TemporalType.DATETIME.

**Comment by suedschwede** [comment]
I would use Datatypes without NULLABLE in Superset

NULLABLE(DateTime) = DateTime

I use as workaround  (clickhouse.py)

```

@classmethod
    def get_datatype(cls, type_code: Any) -> Optional[str]:
        """
        Change column type code from cursor description to string representation.

        :param type_code: Type code from cursor description
        :return: String representation of type code
        """
         
        type_code = type_code.upper()
 
        if (type_code.find('NULLABLE(') != -1): 
          type_code = type_code.replace("NULLABLE(", "")
          type_code = type_code.replace(")", "",1)


        if isinstance(type_code, str) and type_code != "":
            return type_code.upper()
        return None

```
